sympy/polys/polyclasses.py

```diff
@@ -1949,14 +1949,19 @@ def _pexquo(f, g):
         """Polynomial exact pseudo-quotient of ``f`` and ``g``. """
         d = f.degree() - g.degree() + 1
         q, r = divmod(g.LC()**d * f._rep, g._rep)
-        if not r:
+        if r:
             raise ExactQuotientFailed(f, g)
-        return q
+        return f.from_rep(q, f.dom)
 
     def _div(f, g):
         """Polynomial division with remainder of ``f`` and ``g``. """
-        q, r = divmod(f._rep, g._rep)
-        return f.from_rep(q, f.dom), f.from_rep(r, f.dom)
+        if f.dom.is_Field:
+            q, r = divmod(f._rep, g._rep)
+            return f.from_rep(q, f.dom), f.from_rep(r, f.dom)
+        else:
+            # XXX: python-flint defines division in ZZ[x] differently
+            q, r = f.to_DMP_Python()._div(g.to_DMP_Python())
+            return q.to_DUP_Flint(), r.to_DUP_Flint()
```

Comment:
In python-flint the division of polynomials in `ZZ[x]` is defined differently from in sympy:
```python
In [11]: f = flint.fmpz_poly([3, 10, 5]) # flint

In [12]: g = flint.fmpz_poly([2, 2])

In [13]: f
Out[13]: 5*x^2 + 10*x + 3

In [14]: g
Out[14]: 2*x + 2

In [15]: divmod(f, g)
Out[15]: (2*x + 3, x^2 + (-3))

In [6]: f = 5*x**2 + 10*x + 3 # sympy

In [7]: g = 2*x + 2

In [8]: div(f, g, domain=ZZ)
Out[8]:
⎛      2           ⎞
⎝0, 5⋅x  + 10⋅x + 3⎠
```
I don't think that there is an especially good definition. At least the behaviour should be consistent with or without python-flint so we preserve existing behaviour by still using the existing implementation.